inc/WpdbStorage/DataSourceCrud.php

```diff
@@ -37,7 +37,19 @@ public static function get_config_by_uuid( string $uuid ): array|WP_Error {
 	}
 
 	public static function get_configs(): array {
-		return self::get_all_configs();
+		return array_map(
+			function ( array $config ) {
+				// Inflate the config to check if it's valid.
+				$instance = self::inflate_config( $config );
+
+				// If the data source is valid, set a transient field called enabled to true or false otherwise.
+				$config['enabled'] = ! is_wp_error( $instance );
```

Comment:
I see any easy path to displaying errors:

```suggestion
				$config['errors'] = is_wp_error( $instance ) ? [ $instance ] : [];
```

Reply:
I had gone back and forth between doing it this way, or having both `enabled` as well as `errors`. I went for the latter, with `errors` to be added in a later PR that also shows it as a notice on the settings screen UI.

I'll change it up in this PR itself.